mono/mini/driver.c

```diff
@@ -1711,6 +1697,23 @@ mono_enable_counters (void)
 	mono_atomic_store_bool (&mono_jit_stats.enabled, TRUE);
 }
 
+static MonoMethodDesc *
+parse_qualified_method_name (char *method_name)
+{
+	if (strlen (method_name) == 0) {
+		g_printf ("Couldn't parse empty method name.");
+		exit (1);
+	}
+	if (mono_stats_method_desc)
+		g_free (mono_stats_method_desc);
```

Comment:
I think this is the wrong place to do the free. This is just the static function you've written to parse in the name - the free needs to be at the same place where you set the value to the newly parsed method. This ensures that you only have to take the lock once, for as short a time as possible, and makes obvious to someone else reading the code that you really are freeing the old value.